### PR TITLE
Nikhil_Assignment_18_Done_(180+20+5+5)210Min

### PR DESCRIPTION
**Pull request by NikhilMJun20**
->Completed and pushed Assignment # 18
->Please review and comment.

**Note:**
I am not at all happy about the time that I took for Program # 1. 
After spending almost 3 hours, I could come up with a solution which I don't find elegant at all. 
Please suggest better solution and help me improve my timing. 

->PFB screenshots of the outputs of the programs:-

**1** 
Output_WordFrequency

**2** 
Output_MiddleCharacter

**3**
Output_SpecialCharCount

**4** 
Output_StringReverse

### CODE REVIEW

**Comment by NikhilMJun20** [comment]
->Made improvement and changes as suggested in the solution session.
->Please review, comment and approve.

File: src/nikhilMJun20/MiddleCharacter.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+public class MiddleCharacter {
+	void displayMiddleChar(String[] strArray) {
+		int midCharIndex = 0;
+		for(int index=0; index<strArray.length; index++) {
+			if(strArray[index].length()%2 == 0)
+				midCharIndex = (strArray[index].length())/2 - 1;
+			else
+				midCharIndex = (strArray[index].length())/2;
+			System.out.println("Middle chracter of String "+strArray[index]+": "+strArray[index].charAt(midCharIndex));
```
Comment (KrishnaKTechnocredits) [inline src/nikhilMJun20/MiddleCharacter.java:18]: Good job.

File: src/nikhilMJun20/SpecialCharCount.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		str1 = str1.replace('{', ' ');
+		str1 = str1.replace(']', ' ');
+		str1 = str1.replace('}', ' ');
+		str1 = str1.replace('_', ' ');
+		str1 = str1.replace(',', ' ');
+		
+		String[] strArray = str1.split(" ");
+		System.out.println("\nNo. of Special Characters: "+(strArray.length - 1));*/
+		
+		System.out.println("Input String: "+str+"\nNo. of Special Characters in the String = "+(str.length() - str.replace("[", "").replace("{", "").replace("]", "").replace("}", "").replace("_", "").replace(",", "").length()));
```
Comment (KrishnaKTechnocredits) [inline src/nikhilMJun20/SpecialCharCount.java:24]: Good job.

File: src/nikhilMJun20/StringReverse.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+Output : stiderconhcet si siht olleH
+*/
+
+package nikhilMJun20;
+
+public class StringReverse {
+	void displayReversedString (String str) {
+		System.out.println("Given String:\n"+str+"\n\nReversed String:- ");
+		for(int index=str.length()-1; index>=0; index--)
+			System.out.print(str.charAt(index));
```
Comment (KrishnaKTechnocredits) [inline src/nikhilMJun20/StringReverse.java:13]: Pefect.

File: src/nikhilMJun20/WordFrequency.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+public class WordFrequency {
+	void displayWordFreq(String str) {
+		str = str.toLowerCase().replace("_", "");
+		String[] strArray = str.split(" ");
+		int wordCnt = 0;
+		
+		System.out.println("Input String: "+str+"\n\nFrequency of Every Word:-");
+		for(int index=0; index<strArray.length; index++) {
+			wordCnt=1;
+			if(!strArray[index].equals("*")) {
```
Comment (KrishnaKTechnocredits) [inline src/nikhilMJun20/WordFrequency.java:25]: Good job.

**Review by KrishnaKTechnocredits — APPROVED** [review]
@Nikhil you forget to add Screenshot, np.

**Comment by nikhil** [comment]
Sorry about that

**Comment by nikhil** [comment]
I think you tagged the wrong person

**Pull request merged**